expression/integration_serial_test.go

```diff
@@ -3736,8 +3736,8 @@ func TestSetVariables(t *testing.T) {
 	tk.MustExec("set global tidb_enable_noop_functions=1")
 
 	_, err = tk.Exec("set @@global.max_user_connections='';")
-	require.NoError(t, err)
-	//require.Error(t, err, variable.ErrWrongTypeForVar.GenWithStackByArgs("max_user_connections").Error())
+	require.Error(t, err)
+	require.Error(t, err, variable.ErrWrongTypeForVar.GenWithStackByArgs("max_user_connections").Error())
```

Comment:
So it's still a compatibility breaker? I'll add a label.

Reply:
Sure. We can add the label, but it is not very likely since this was only introduced in https://github.com/pingcap/tidb/pull/31566 (5.4) - This PR reverts the previous PR with one that handles noops *and* other sysvars.